src/services/utils.ts

```diff
@@ -31,6 +31,68 @@ const parseError = (text: string) => {
     }
 };
 
+export type BackendProblemDetail = {
+    status?: number;
+    businessErrorCode?: string;
+};
+
+type SpringErrorPayload = {
+    status?: number;
+    message?: string;
+};
+
+export interface CustomError extends Error {
```

Comment:
This should not be just an interface, it should be a class otherwise you cannot do instanceof